packages/utils/pack-up/src/node/tasks/dts/diagnostic.ts

```diff
@@ -0,0 +1,42 @@
+import chalk from 'chalk';
+import path from 'path';
+import ts from 'typescript';
+
+import { Logger } from '../../core/logger';
+
+const printDiagnostic = (
+  diagnostic: ts.Diagnostic,
+  { logger, cwd }: { logger: Logger; cwd: string }
+) => {
+  if (diagnostic.file && diagnostic.start) {
+    const { line, character } = ts.getLineAndCharacterOfPosition(diagnostic.file, diagnostic.start);
+    const message = ts.flattenDiagnosticMessageText(diagnostic.messageText, ts.sys.newLine);
+
+    const file = path.relative(cwd, diagnostic.file.fileName);
+
+    const output = [
```

Comment:
Out of scope here, but just a comment that we're starting to add more and more custom loggers (this and data-transfer come to mind as things that different from the strapi server logger, and CLI in general doesn't have any standards) and we should start looking for some universal standards to apply everywhere.

Reply:
yeah it would be nice to do that. Probably should live in `utils` or something, I guess it's more CLI work than server work?

Reply:
Yeah, maybe we'll need a cli-utils, would be quite useful to share some code between our various CLIs. For example, I was going to point out the existence of the expectExit function (currently duplicated in two places from the ts migration) that would let you test the process.exit codes of your cli tests.

I'll think about the idea and maybe put something together later.